### PR TITLE
samples: matter: Removed using relative paths in CMakeLists.txt

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
The matter samples use in CMakeLists.txt relative paths from its
location in order to add some source files and configs. That makes
a problem when samples are extracted as templates to other locations
than original ones.

Changes:
* Replaced relative paths to common directory with the path using
${ZEPHYR_NRF_MODULE_DIR}
* Replaced inserting overlay-dfu_support.conf by using prj_dfu.conf

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: samples/matter/lock/README.rst [hunk 1]
```diff
@@ -82,12 +82,12 @@ Device Firmware Upgrade support
 
 You can configure the sample to use the secure bootloader for performing over-the-air Device Firmware Upgrade using Bluetooth® LE, using the following build flags during the build process:
 
-* ``-DOVERLAY_CONFIG=../common/config/overlay-dfu_support.conf``
```
Comment (Damian-Nordic) [inline samples/matter/lock/README.rst:85]: Why doesn't the doc recommend using BUILD_WITH_DFU option?

Comment (kkasperczyk-no) [reply samples/matter/lock/README.rst:85]: Somehow it does below, but I'm also not sure why it was changed. It seems that the change was introduced in https://github.com/nrfconnect/sdk-nrf/pull/5157. @FrancescoSer do you happen to remember why it was done like this, as for me we should recommend easier option, but I might not have full view of a problem.

Comment (FrancescoSer) [reply samples/matter/lock/README.rst:85]: The scope of PR 5157 was to separate the list of the flags that needed to be set for running the build from the example shown, as it uses cmd and not SES. There might have been an issue in this specific case.

Just to clarify, which are the correct flags required for building with DFU in this case?

Comment (kkasperczyk-no) [reply samples/matter/lock/README.rst:85]: > Just to clarify, which are the correct flags required for building with DFU in this case? If there is an issue there we can fix that in an additional commit.

Actually both of them, but inserting `-- -DCONF_FILE=prj_dfu.conf -DPM_STATIC_YML_FILE="configuration/build-target/pm_static.yml` was not user friendly, so we added `-- -DBUILD_WITH_DFU=1` option that basically does the same under the hood.

**Review by LuDuda — APPROVED** [review]
Looks good.

This change, unfortunately, increases the number of configuration files in the samples, though anyway, we're going to enable DFU (over Matter) by default probably from the next release, which will reduce them again, and we would need only simple overlay to enable SMP.

File: samples/matter/lock/README.rst (hunk 1 above)
Comment (kkasperczyk-no) [reply samples/matter/lock/README.rst:85]: @FrancescoSer I think that -DBUILD_WITH_DFU=1 flag should be also possible to add in SES, so I modified description a little bit to use simplified form and single option, but also left reference to setting cmake options instruction. Please verify if you are ok with that.

**Review by Damian-Nordic — APPROVED** [review]

File: samples/matter/lock/README.rst (hunk 1 above)
Comment (greg-fer) [reply samples/matter/lock/README.rst:85]:
```suggestion
You can configure the sample to use the secure bootloader for performing over-the-air Device Firmware Upgrade using Bluetooth® LE, using the ``-DBUILD_WITH_DFU=1`` build flag during the build process:
```
Then we also don't need line 85...

File: samples/matter/lock/README.rst [hunk 2]
```diff
@@ -80,14 +80,13 @@ Device Firmware Upgrade support
 .. note::
    Over-the-air Device Firmware Upgrade can be enabled only on hardware platforms containing external flash memory.
 
-You can configure the sample to use the secure bootloader for performing over-the-air Device Firmware Upgrade using Bluetooth® LE, using the following build flags during the build process:
+You can configure the sample to use the secure bootloader for performing over-the-air Device Firmware Upgrade using Bluetooth® LE, using the following build flag during the build process:
 
-* ``-DOVERLAY_CONFIG=../common/config/overlay-dfu_support.conf``
```
Comment (greg-fer) [inline samples/matter/lock/README.rst:85]:
```suggestion
You can configure the sample to use the secure bootloader for performing over-the-air Device Firmware Upgrade using Bluetooth® LE, using the ``-DBUILD_WITH_DFU=1`` build flag during the build process:
```

**Review by greg-fer — APPROVED** [review]

**Comment by greg-fer** [comment]
No changelog entry needed?

**Comment by kkasperczyk-no** [comment]
@greg-fer I don't think so, those changes are related to the building process.

**Pull request merged**